src/board.rs

```diff
@@ -108,73 +111,76 @@ impl Board {
         self.add_moves();
     }
 
-    fn pawn_can_double_move(&self, position: Position, player: Player) -> bool {
-        let v = Displacement::get_pawn_advance_vector(player);
-        if let None = self.get_piece(position + v * 2) {
-            return match self.get_piece(position).unwrap().get_player() {
-                Player::White => position.y == 1,
-                Player::Black => position.y == 6,
-            };
-        }
-        false
-    }
-
-    fn add_pawn_advance_moves(&mut self, start: Position, player: Player) {
+    fn add_pawn_advance_moves(&mut self, from: &Position, player: Player) {
         let v = Displacement::get_pawn_advance_vector(player);
-        let new_position = start + v;
-        if Self::is_in_bounds(new_position).is_ok() && self.get_piece(new_position).is_none() {
+        let mut to = from.clone() + v;
```

Comment:
So I kinda think the rule should be that if you have to clone a variable in a function anyway, it should be passed by value (when `Copy` is implemented). Thoughts?